### PR TITLE
ER-1106 Withdraw live vacancy +semver: Minor

### PR DESCRIPTION
**Pull request by mrpmorris**

### CODE REVIEW

File: src/QA/QA.Web/Controllers/WithdrawVacancyController.cs [hunk 1]
```diff
@@ -27,28 +27,37 @@ public WithdrawVacancyController(WithdrawVacancyOrchestrator orchestrator)
...
             var vm = _orchestrator.GetFindVacancyViewModel();
+            vm.VacancyReference = TempData[TempDataKeys.WithdrawVacancyReference]?.ToString();
 
             return View(vm);
         }
 
         [HttpPost(Name = RouteNames.WithdrawVacancy_FindVacancy_Post)]
         public async Task<IActionResult> FindVacancy(FindVacancyEditModel m)
         {
+            string userInputVacancyReference = m?.VacancyReference;
```
Comment (sachpatel) [inline src/QA/QA.Web/Controllers/WithdrawVacancyController.cs]: The model will not be null though so I don't think you need this null check

File: src/QA/QA.Web/Models/WithdrawVacancy/PostFindVacancyEditModelResult.cs [hunk 2]
```diff
@@ -4,7 +4,8 @@ public enum PostFindVacancyEditModelResultType
     {
         NotFound,
         AlreadyClosed,
-        CanClose
+        CanClose,
+        CannotClose
```
Comment (dev4valtech) [inline src/QA/QA.Web/Models/WithdrawVacancy/PostFindVacancyEditModelResult.cs]: Do we need to differentiate between cannot close and already closed? Perhaps the messages are different in which case, can you rename cannot close to NotLive?

File: src/QA/QA.Web/Views/Dashboard/Index.cshtml [hunk 3]
```diff
@@ -139,7 +139,10 @@
         <p class="govuk-body">Create and download reports for your vacancies for any period of time.</p>
     </div>
     
-    <div asp-hide="true"><!-- Temporarily hide this for public beta release -->
+    @{ 
+        //TODO: PeteM - set asp-hide="true"
```
Comment (dev4valtech) [inline src/QA/QA.Web/Views/Dashboard/Index.cshtml]: Remove asp-hide altogether and the TODO comment

File: src/QA/QA.Web/Controllers/WithdrawVacancyController.cs [hunk 4]
```diff
@@ -27,31 +27,39 @@ public WithdrawVacancyController(WithdrawVacancyOrchestrator orchestrator)
...
+            TempData[TempDataKeys.WithdrawVacancyReference] = userInputVacancyReference;
+
             if (ModelState.IsValid)
             {
                 var result = await _orchestrator.PostFindVacancyEditModelAsync(m);
 
-                TempData[TempDataKeys.WithdrawVacancyReference] = result.VacancyReference;
+                long? retrievedVacancyReference = result.VacancyReference;
```
Comment (dev4valtech) [inline src/QA/QA.Web/Controllers/WithdrawVacancyController.cs]: This doesn't seem to be used anywhere.

File: src/QA/QA.Web/Controllers/WithdrawVacancyController.cs [hunk 5]
```diff
@@ -151,8 +159,16 @@ public async Task<IActionResult> Acknowledgement()
 
         private long? GetVacancyReference()
         {
-            if (long.TryParse(TempData.Peek(TempDataKeys.WithdrawVacancyReference)?.ToString(), out long vacancyReferenceNumber))
+            string userInputVacancyReference = (string)TempData.Peek(TempDataKeys.WithdrawVacancyReference);
+            //Store the initial input value in case the user goes back to the previous screen
+            TempData[TempDataKeys.WithdrawVacancyReference] = userInputVacancyReference;
```
Comment (dev4valtech) [inline src/QA/QA.Web/Controllers/WithdrawVacancyController.cs]: If you are peeking, you don't need to store the temp data back.

**Pull request closed without merging**